Firestore/Example/Tests/Local/FSTMutationQueueTests.mm

```diff
@@ -296,75 +300,75 @@ - (void)testAllMutationBatchesAffectingQuery {
     NSMutableArray<FSTMutationBatch *> *batches = [NSMutableArray array];
     for (FSTMutation *mutation in mutations) {
       FSTMutationBatch *batch =
-          [self.mutationQueue addMutationBatchWithWriteTime:[FIRTimestamp timestamp]
-                                                  mutations:@[ mutation ]];
+          self.mutationQueue->AddMutationBatch([FIRTimestamp timestamp], @[ mutation ]);
       [batches addObject:batch];
     }
 
-    NSArray<FSTMutationBatch *> *expected = @[ batches[1], batches[2], batches[4] ];
+    std::vector<FSTMutationBatch *> expected = {batches[1], batches[2], batches[4]};
     FSTQuery *query = FSTTestQuery("foo");
-    NSArray<FSTMutationBatch *> *matches =
-        [self.mutationQueue allMutationBatchesAffectingQuery:query];
+    std::vector<FSTMutationBatch *> matches =
+        self.mutationQueue->AllMutationBatchesAffectingQuery(query);
 
-    XCTAssertEqualObjects(matches, expected);
+    [self assertVector:matches matchesExpected:expected];
   });
 }
 
 - (void)testRemoveMutationBatches {
   if ([self isTestBaseClass]) return;
 
   self.persistence.run("testRemoveMutationBatches", [&]() {
-    NSMutableArray<FSTMutationBatch *> *batches = [self createBatches:10];
+    std::vector<FSTMutationBatch *> batches = [self createBatches:10];
 
-    [self.mutationQueue removeMutationBatch:batches[0]];
-    [batches removeObjectAtIndex:0];
+    self.mutationQueue->RemoveMutationBatch(batches[0]);
+    batches.erase(batches.begin());
 
     XCTAssertEqual([self batchCount], 9);
 
-    NSArray<FSTMutationBatch *> *found;
+    std::vector<FSTMutationBatch *> found;
 
-    found = [self.mutationQueue allMutationBatches];
-    XCTAssertEqualObjects(found, batches);
-    XCTAssertEqual(found.count, 9);
+    found = self.mutationQueue->AllMutationBatches();
+    [self assertVector:found matchesExpected:batches];
+    XCTAssertEqual(found.size(), 9);
 
-    [self.mutationQueue removeMutationBatch:batches[0]];
-    [self.mutationQueue removeMutationBatch:batches[1]];
-    [self.mutationQueue removeMutationBatch:batches[2]];
-    [batches removeObjectsInRange:NSMakeRange(0, 3)];
+    self.mutationQueue->RemoveMutationBatch(batches[0]);
+    self.mutationQueue->RemoveMutationBatch(batches[1]);
+    self.mutationQueue->RemoveMutationBatch(batches[2]);
+    for (int i = 0; i < 3; i++) {
+      batches.erase(batches.begin());
+    }
```

Comment:
Alternative:

batches.erase(batches.begin(), batches.begin() + 3);

which I think also matches the original a bit better.

Reply:
Done, although using math on iterators like that scares me a little.